sentinel.go

```diff
@@ -319,13 +339,28 @@ func (c *sentinelClient) listWatch(cc conn) (master string, sentinels []string,
 				if m[0] == "master" && m[1] == c.sOpt.Sentinel.MasterSet {
 					c.switchMasterRetry(fmt.Sprintf("%s:%s", m[2], m[3]))
 				}
+			case "+slave":
```

Comment:
Should we subscribe to `-slave` as well?

Reply:
I couldn't find this channel in official sentinel documentation ([here](https://redis.io/docs/management/sentinel/#pubsub-messages)). Also removing a slave replica locally doesn't seem to trigger it. Although `+sdown` and `-sdown` is triggered when a replica (or master) is restarted and we can capture it with condition `m[0] == "slave"`.

Reply:
I am sorry about the misinformation. What about failover-end?

Reply:
I think the only condition that's not covered by current channels is the condition of `slave` shutting down. master restart/shutdowns are covered with `+switch-master` and `+reboot`. Also `sdown` status of replicas isn't reported by `failover-end` as I've tested. 

What do you think about the changes proposed in ab120af1ad7695448e6bc7e718f9a045820da152 and 37741dd8065a9601b4c33c5209ff70545d4b9bb4?

Feel free to check your other comments and check if they're resolved.

Reply:
Monitoring `+sdown` and `-sdown` looks good to me.

Just one concern about checking the existence of `s-down-time`: Will it be clear once the replica is alive?

Besides that, I think the way you `delete response i` may panic: https://go.dev/play/p/zWeXCrTNqwJ

Reply:
Thanks for the precise review :)

Sorry, I've updated the method to avoid panic.

Also `s-down-time` seems to disappear once the replica is live and can reply to sentinel `PING`s. I couldn't find any documentation on it to confirm but I've tested it with multiple scenarios and it looks like to work that way.